.github/scripts/rust/lint.rs

```diff
@@ -0,0 +1,255 @@
+//! ```cargo
+//! [dependencies]
+//! serde = {version = "*", features = ["derive"]}
+//! toml = {version = "*", features = ["parse"]}
+//! ```
+use serde::{Deserialize, Serialize};
+use std::env;
+use std::fs;
+use std::collections::{HashMap, HashSet};
+use std::path::{PathBuf, Path};
+use std::iter::once;
+
+enum LintLevel {
+    Allow,
+    Warn,
+    Deny,
+    Forbid,
+}
+
+struct Lints(HashMap<String, LintLevel>);
+
+impl Lints {
+    pub fn new() -> Self {
+        Self(HashMap::new())
+    }
+
+    // if defined multiple times this is the priority order
+    // * allow
+    // * warn
+    // * deny
+    // * forbid
+    pub fn apply(&mut self, file: LintFile) {
+        for lint in file.allow {
+            self.0.insert(lint, LintLevel::Allow);
+        }
+
+        for lint in file.warn {
+            self.0.insert(lint, LintLevel::Warn);
+        }
+
+        for lint in file.deny {
+            self.0.insert(lint, LintLevel::Deny);
+        }
+
+        for lint in file.forbid {
+            self.0.insert(lint, LintLevel::Forbid);
+        }
+
+        for lint in file.default {
+            self.0.remove(&lint);
+        }
+    }
+
+    pub fn into_file(self) -> LintFile {
+        self.0.into_iter().fold(LintFile::default(), |mut acc, (lint, level)| {
+            match level {
+                LintLevel::Allow => acc.allow.insert(lint),
+                LintLevel::Warn => acc.warn.insert(lint),
+                LintLevel::Deny => acc.deny.insert(lint),
+                LintLevel::Forbid => acc.forbid.insert(lint),
+            };
+
+            acc
+        })
+    }
+}
+
+#[derive(Deserialize, Default)]
+struct LintFile {
+    #[serde(default)]
+    allow: HashSet<String>,
+    #[serde(default)]
+    warn: HashSet<String>,
+    #[serde(default)]
+    deny: HashSet<String>,
+    #[serde(default)]
+    forbid: HashSet<String>,
+    #[serde(default)]
+    default: HashSet<String>,
+}
```

Comment:
`force-warn` is missing but I don't know if there is a flag for it

Reply:
There is, `--force-warn`, will add.

Reply:
fixed in [6296e8e](https://github.com/hashintel/hash/pull/2075/commits/6296e8e0910adbf20de2627ac7966415a999b836)